### PR TITLE
Makes flashers actually flash again!

### PR DESCRIPTION
**Pull request by ShizCalev**
:cl: ShizCalev
fix: Wall mounted & portable flashers now actually flash again!
admin: Flashers can now have their range and cooldown durations vareditted with flash_range and flash_cooldown_duration respectively.
/:cl:

Not sure how long they've been broken tbh. 

Switched them over from doing some weird funky lighting system memes (that didn't work since there's mixed lighting systems) to just using the `flash_lighting_fx()` proc that I made over 4 years ago to literally handle doing this exact thing.

Rest is just code cleanup + better varediting support.

### CODE REVIEW

File: code/game/machinery/flasher.dm [hunk 1]
```diff
@@ -8,40 +8,45 @@
...
 	var/id = null
-	var/range = 2 //this is roughly the size of brig cell
-	var/last_flash = 0 //Don't want it getting spammed like regular flashes
-	var/strength = 100 //How knocked down targets are when flashed.
+	/// How far this flash reaches. Affects both proximity distance and the actual stun effect.
+	var/flash_range = 2 //this is roughly the size of a brig cell.
 
-MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/flasher, 26)
+	/// How strong Paralyze()'d targets are when flashed.
+	var/strength = 100
```
Comment (Fikou) [inline code/game/machinery/flasher.dm:19]: if this is passed in directly as an argument for time make it use seconds defines

File: code/game/machinery/flasher.dm [hunk 2]
```diff
@@ -97,24 +100,23 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/flasher, 26)
...
-	set_light_on(TRUE)
-	addtimer(CALLBACK(src, .proc/flash_end), FLASH_LIGHT_DURATION, TIMER_OVERRIDE|TIMER_UNIQUE)
+	flash_lighting_fx()
 
-	last_flash = world.time
+	COOLDOWN_START(src, flash_cooldown, flash_cooldown_duration)
 	use_power(1000)
 
 	var/flashed = FALSE
 	for (var/mob/living/L in viewers(src, null))
```
Comment (Fikou) [inline code/game/machinery/flasher.dm:118]: while youre touching this, viewers is only mobs i think? so you can use as anything, and change the name from L

File: code/game/machinery/flasher.dm (hunk 1 above)
Comment (ShizCalev) [reply code/game/machinery/flasher.dm:19]:
```suggestion
	var/strength = 10 SECONDS
```

File: code/game/machinery/flasher.dm (hunk 2 above)
Comment (ShizCalev) [reply code/game/machinery/flasher.dm:118]: that'd include camera mobs & ghosts.

File: code/game/machinery/flasher.dm [hunk 3]
```diff
@@ -160,19 +156,35 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/flasher, 26)
 			new /obj/item/stack/sheet/iron (loc, 2)
 	qdel(src)
 
+/obj/machinery/flasher/portable //Portable version of the flasher. Only flashes when anchored
+	name = "portable flasher"
+	desc = "A portable flashing device. Wrench to activate and deactivate. Cannot detect slow movements."
+	icon_state = "pflash1-p"
+	base_icon_state = "pflash"
+	strength = 80
```
Comment (Fikou) [inline code/game/machinery/flasher.dm:164]: here too btw

File: code/game/machinery/flasher.dm (hunk 2 above)
Comment (Fikou) [reply code/game/machinery/flasher.dm:118]: just change the name then

**Pull request merged**